clang compilation warning/error

When compiling with clang, I get a warning (by default promoted to an error):
```
hfile_irods.c:149:23: warning: address of array 'irods.env.rodsHost' will always evaluate to 'true' [-Wpointer-bool-conversion]
        if (irods.env.rodsHost && irods.env.rodsPort)
            ~~~~~~~~~~^~~~~~~~ ~~
```

--- a/hfile_irods.c
+++ b/hfile_irods.c
@@ -146,7 +146,7 @@ static int irods_init()
         fprintf(stderr, "[M::hfile_irods.init] connected to %s(%s)",
                 irods.conn->svrVersion->relVersion,
                 irods.conn->svrVersion->apiVersion);
-        if (irods.env.rodsHost && irods.env.rodsPort)
+        if (strncmp(irods.env.rodsHost, "", 1) != 0 && irods.env.rodsPort)
             fprintf(stderr, " at %s:%d\n",
                     irods.env.rodsHost, irods.env.rodsPort);
         else

